ginivision/src/main/java/net/gini/android/vision/camera/ImageStack.java

```diff
@@ -0,0 +1,308 @@
+        final Drawable drawable1 = stackItem1.getDrawable();
+        final Drawable drawable2 = stackItem2.getDrawable();
+        final Drawable drawable3 = stackItem3.getDrawable();
+
+        // Prepare the transition listeners
+        addImageTransitionListener.setDrawable1(drawable1);
+        addImageTransitionListener.setDrawable2(drawable2);
+        addImageTransitionListener.setDrawable3(drawable3);
+        addImageTransitionListener.setNewImage(bitmap);
+
+        cleanupTransitionListener.setDrawable1(drawable1);
+        cleanupTransitionListener.setDrawable2(drawable2);
+        cleanupTransitionListener.setNewImage(bitmap);
+
+        // Execute the transition
+        transitionManager.transitionTo(imageAddedScene);
+    }
+
+    public void removeImages() {
+        setImages(null);
+    }
+
+    public void setImages(@Nullable final List<Bitmap> bitmaps) {
+        imageCount = 0;
+        stackItem1.setImageDrawable(null);
+        stackItem1.setClickable(false);
+        stackItem1.setFocusable(false);
+        stackItem2.setImageDrawable(null);
+        stackItem3.setImageDrawable(null);
+        badge.setText("");
+        badge.setVisibility(INVISIBLE);
+        if (bitmaps == null) {
+            return;
+        }
+        imageCount = bitmaps.size();
+        if (imageCount > 0) {
+            badge.setVisibility(View.VISIBLE);
+            badge.setText(String.valueOf(imageCount));
+        }
+        if (bitmaps.size() > 0) {
+            stackItem1.setClickable(true);
+            stackItem1.setFocusable(true);
+            if (clickListener != null) {
+                stackItem1.setOnClickListener(clickListener);
+            }
+        }
+        if (bitmaps.size() > 2) {
+            stackItem3.setImageBitmap(bitmaps.get(0));
+            stackItem2.setImageBitmap(bitmaps.get(1));
+            stackItem1.setImageBitmap(bitmaps.get(2));
+        } else if (bitmaps.size() > 1) {
+            stackItem2.setImageBitmap(bitmaps.get(0));
+            stackItem1.setImageBitmap(bitmaps.get(1));
+        } else if (bitmaps.size() > 0) {
+            stackItem1.setImageBitmap(bitmaps.get(0));
+        }
+    }
+
+    private static class AddImageTransitionListener extends TransitionListenerAdapter {
+
+        private final Scene imageAddedScene;
+        private final ImageStack imageStack;
+        private Drawable drawable1;
+        private Drawable drawable2;
+        private Drawable drawable3;
+        private Bitmap newImage;
+
+        private AddImageTransitionListener(final Scene imageAddedScene,
+                final ImageStack imageStack) {
+            this.imageAddedScene = imageAddedScene;
+            this.imageStack = imageStack;
+        }
+
+        @Override
+        public void onTransitionStart(@NonNull final Transition transition) {
+            final ViewGroup sceneRoot = imageAddedScene.getSceneRoot();
+            final ImageView stackItem1View = sceneRoot.findViewById(R.id.gv_stack_item_1);
```

Comment:
The views have to be fetched on every animation start. I tried keeping the references but the views weren't update on subsequent transitions.